metadata/src/main/java/edu/unc/lib/dl/util/PremisEventBuilder.java

```diff
@@ -0,0 +1,100 @@
+package edu.unc.lib.dl.util;
+
+import java.text.MessageFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import com.hp.hpl.jena.rdf.model.Model;
+import com.hp.hpl.jena.rdf.model.ModelFactory;
+import com.hp.hpl.jena.rdf.model.Resource;
+
+import edu.unc.lib.dl.event.PremisLogger;
+import edu.unc.lib.dl.rdf.Premis;
+
+public class PremisEventBuilder {
+	private String eventId;
+	private Model model;
+	private PremisLogger premisLogger;
+	private Resource premisObjResc;
+	
+	public PremisEventBuilder(String eventId, Resource eventType, Date date,
+			PremisLogger premisLogger) {
+		this.eventId = eventId;
+		this.premisLogger = premisLogger;
+		addEvent(eventType, date);
+	}
+
+	private PremisEventBuilder addEvent(Resource eventType, Date date) {
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
+		Resource premisObjResc = getResource();
+
+		premisObjResc.addProperty(Premis.hasEventType, eventType);
+		premisObjResc.addProperty(Premis.hasEventDateTime, dateFormat.format(date));
+		
+		return this;
+	}
+	
+	public PremisEventBuilder addEventDetail(String message, Object... args) {
+		if(args != null) {
```

Comment:
Might as well not persist the squished "if(" into a new class.  There are a couple in here